test/pytest/test_bridge.py

```diff
@@ -419,6 +419,42 @@ async def test_fsread1_errors(transport):
                                reply_keys={'message': "attribute 'max_read_size': must have type int"})
 
 
+@pytest.mark.asyncio
+async def test_fsread1_size_hint(transport):
+    transport.send_open('fsread1', path='/etc/passwd', binary='raw')
```

Comment:
You should just use `transport.check_open()` with `reply_keys={'size-hint: buf.st_size}`.

Reply:
I'd also try to read the data and make sure you get the correct number of bytes, then a done and a close.  That will leave you in a better state stat the start of the next case (and you won't have to do the loop to ignore unrelated messages).  Alternatively, you could just put each test into its own separate case — that's not an unusual approach with pytest.